mypy/checkmember.py

```diff
@@ -778,32 +783,39 @@ def analyze_class_attribute_access(itype: Instance,
         if not mx.is_lvalue:
             result = analyze_descriptor_access(mx.original_type, result, mx.builtin_type,
                                                mx.msg, mx.context, chk=mx.chk)
-        return result
+
+        return apply_class_attr_hook(mx, hook, result)
     elif isinstance(node.node, Var):
+        # TODO RIGHT NOW NOMERGE - is it okay to not modify this?
         mx.not_ready_callback(name, mx.context)
         return AnyType(TypeOfAny.special_form)
 
     if isinstance(node.node, TypeVarExpr):
+        # TODO RIGHT NOW NOMERGE - is it okay to not modify this?
         mx.msg.fail(message_registry.CANNOT_USE_TYPEVAR_AS_EXPRESSION.format(
                     info.name, name), mx.context)
         return AnyType(TypeOfAny.from_error)
 
     if isinstance(node.node, TypeInfo):
-        return type_object_type(node.node, mx.builtin_type)
+        assert False, "TODO RIGHT NOW NOMERGE - No tests hit this, how to trigger?"
+        return apply_class_attr_hook(mx, hook, type_object_type(node.node, mx.builtin_type))
 
     if isinstance(node.node, MypyFile):
         # Reference to a module object.
-        return mx.builtin_type('types.ModuleType')
+        assert False, "TODO RIGHT NOW NOMERGE - No tests hit this, how to trigger?"
+        return apply_class_attr_hook(mx, hook, mx.builtin_type('types.ModuleType'))
```

Comment:
We shouldn't run the hook here, since it wouldn't get triggered consistently. Some references to module attributes are resolved in semantic analysis, before type checking. The hook should be applied either always for never for each particular attribute.